### PR TITLE
Handles replay tests cleanups

### LINKED ISSUE #1413: [Handles] Recheck replay tests
- [ ]  Check over replay tests and make sure they are testing the failure case when inside of 100 blocks. test_handle_claim_replay_attack does not. Some of these might be duplicates. See pallets/handles/src/tests/handles_replay_attack_test.rs
- [ ] ?

### PR DESCRIPTION
**Pull request by saraswatpuneet**
# Goal
The goal of this PR is to cleanup replay tests and with better naming 

Closes #1413 

# Checklist
- [ ] Chain spec updated
- [ ] Custom RPC OR Runtime API added/changed? Updated js/api-augment.
- [ ] Design doc(s) updated
- [x] Tests added
- [ ] Benchmarks added
- [ ] Weights updated

### CODE REVIEW

File: pallets/handles/src/tests/handle_retirements_tests.rs [hunk 1]
```diff
@@ -53,6 +54,64 @@ fn claim_and_retire_handle_happy_path() {
...
+			alice.public().into(),
+			proof.clone(),
+			payload.clone()
+		));
+		let msa_id = MessageSourceId::decode(&mut &alice.public().encode()[..]).unwrap();
+		let handle = Handles::get_handle_for_msa(msa_id);
+		assert!(handle.is_some());
+
+		// Compose the full display handle from the base handle, "." delimeter and suffix
+		let handle_result = handle.unwrap();
```
Comment (enddynayn) [inline pallets/handles/src/tests/handle_retirements_tests.rs:77]: i think there is a function here in handles you can use for this.

File: pallets/handles/src/tests/handle_retirements_tests.rs [hunk 2]
```diff
@@ -53,6 +54,64 @@ fn claim_and_retire_handle_happy_path() {
...
+			&RuntimeCall::Handles(HandlesCall::retire_handle {});
+		let info = DispatchInfo::default();
+		let len = 0_usize;
+		let early_retire_result = HandlesSignedExtension::<Test>::new().validate(
+			&alice.public().into(),
+			call_retire_handle,
+			&info,
+			len,
+		);
+		assert!(early_retire_result.is_err());
```
Comment (enddynayn) [inline pallets/handles/src/tests/handle_retirements_tests.rs:96]: Do we need to make sure that we are getting the expected here we want here?

File: pallets/handles/src/tests/handle_retirements_tests.rs (hunk 1 above)
Comment (saraswatpuneet) [reply pallets/handles/src/tests/handle_retirements_tests.rs:77]: That is with the suffix index, at this place we are getting handle from state and forming a display handle (ow a consumer of rpc would do)

File: pallets/handles/src/tests/handle_retirements_tests.rs (hunk 2 above)
Comment (saraswatpuneet) [reply pallets/handles/src/tests/handle_retirements_tests.rs:96]: We are safe from the point of view of that this is a direction function call so it is expected to return error, but I will see if there is a way to do some comparison

**Review by enddynayn — APPROVED** [review]
👍 Nice!

**Review by wilwade — APPROVED** [review]
Much better naming and test. Thanks!

**Pull request merged**